### DISCUSSION TITLE
dependencies in same kustomization

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#2750] [Q&A]
I have a HR that depends on a configmap in the same folder which my kustomization points to, however, on initial install (something we do on testing very often) the kustomization sits in failed state for 10 mins before going healthy as the HR initially couldn't find the ConfigMap.

Is there a way to control order of kustomize applying resources under a single kustomization? Is it based on filename alphabetically or resource alphabetically?

I know we could do multiple folders and then do kustomizations depending on each other but ideally I'd like to try solving without first

**Comment by kingdonb** [comment]
Do you have `spec.wait` set on your Flux Kustomization?

If it's set to `true` this might get worse, if it's set to `false` then it might solve your issue. Can you clarify which setting you have and maybe post the entire Kustomization?

We've had a bit of discussion about this as I think there is an actual UX issue here.

I would suggest adding a setting for `spec.retryInterval` that runs faster than `timeout` or `interval` as another possible mitigation. I guess the 10 minute wait that you're experiencing is correlated to `spec.interval` which defaults to 10 minutes in the CLI. You can then make the interval shorter as another possible mitigation.
